### PR TITLE
➕ Move to TypeScript

### PR DESCRIPTION
**Pull request by frinyvonnick**
## Description

Following the issue #127 and the pull request #132 I decided to give a try to TypeScript.

## Other

I removed unused files and updated the LICENSE

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,18 +13,29 @@
...
   "devDependencies": {
+    "@babel/core": "^7.16.0",
+    "@babel/preset-env": "^7.16.0",
+    "@babel/preset-typescript": "^7.16.0",
+    "@types/handlebars": "^4.1.0",
+    "@types/jest": "^27.0.2",
+    "@types/node": "^16.11.6",
+    "@types/puppeteer": "^5.4.4",
+    "@types/puppeteer-core": "^5.4.0",
+    "babel-jest": "^27.3.1",
```
Comment (Taranys) [inline package.json:33]: I think you can remove `babe-jest` from the dependencies because it is already a dependencies of `jest`. This should simplify the yarn.lock and avoid version conflict :)
```suggestion
```

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { Cluster } from "puppeteer-cluster";
+
+import { Options, Content } from "./types";
```
Comment (Taranys) [inline src/index.ts:3]: The `Content` is not used in the file, you can 🔥 it

```suggestion
import { Options } from "./types";
```

File: src/screenshot.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { Page, Base64ScreenShotOptions } from "puppeteer";
```
Comment (Taranys) [inline src/screenshot.ts:1]: The`Base64ScreenShotOptions` is not used in the file, you can 🔥 it

```suggestion
import { Page } from "puppeteer";
```

File: src/screenshot.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    selector,
+    beforeScreenshot,
+    transparent = false,
+    waitUntil = "networkidle0",
+  }: Options
+) {
+  let screenshotArgs: { quality?: number } = {};
+  if (type === "jpeg") {
+    screenshotArgs.quality = quality ? quality : 80;
+  }
```
Comment (Taranys) [inline src/screenshot.ts:24]: TS is able to infer the type directly if you want
```suggestion
  const screenshotArgs = (type === "jpeg") ? { quality: quality ?? 80 } : {}
```

File: src/screenshot.spec.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
+import { describe } from "jest-circus";
+// @ts-ignore needed for the mock
+import puppeteer from "puppeteer";
```
Comment (Taranys) [inline src/screenshot.spec.ts:3]: Why did you add those two lines ? I tried to run the tests without them and everything seems fine

**Review by Taranys — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (frinyvonnick) [reply package.json:33]: I followed this documentation https://jestjs.io/docs/getting-started#using-typescript which ask to follow this one https://jestjs.io/docs/getting-started#using-babel. It tells to install babel-jest :( 

I'll give it a try 👍

Comment (frinyvonnick) [reply package.json]: Works like a charm 👌

File: src/index.ts (hunk 2 above)
Comment (frinyvonnick) [reply src/index.ts]: I added a linter for this stuff 🙏

File: src/screenshot.ts (hunk 4 above)
Comment (frinyvonnick) [reply src/screenshot.ts]: Fixed, thank you 🙏

File: src/screenshot.spec.ts (hunk 5 above)
Comment (frinyvonnick) [reply src/screenshot.spec.ts]: At some point when I launched the tests without this line one test was failing during the run but not at the end. So it makes me less confident but I can't reproduce it either explain why it did this 🤔

Comment (frinyvonnick) [reply src/screenshot.spec.ts]: I removed it 👌
